infrastructure.md

```diff
@@ -0,0 +1,54 @@
+# CPG Infrastructure
+
+## Dataset
+
+A dataset is one defined collection of resources. It has:
+
+- A GCP project
+- A set of buckets
+- A set of secrets to manage access (access_group_cache, cromwell access keys)
+- A set of allowed_repositories
+
+## Components
+
+- Hail query / batch
+- Analysis-runner
+- Sample-metadata server
```

Comment:
Could you add the repo links here and add a short summary of what these are used for?